insta/src/output.rs

```diff
@@ -103,121 +103,183 @@ impl<'a> SnapshotPrinter<'a> {
+                println!(
+                    "{}",
+                    encode_file_link_escape(
+                        &new_contents
+                            .build_path(self.snapshot_file.unwrap().with_extension("snap.new"))
+                    )
+                );
+            }
         }
-        println!("──────┴{:─^1$}", "", width.saturating_sub(7));
     }
 
     fn print_changeset(&self) {
-        let old: String = self
-            .old_snapshot
-            .map_or("".to_string(), |x| x.contents_string());
-        let new = self.new_snapshot.contents_string();
-        let newlines_matter = newlines_matter(old.as_str(), new.as_str());
-
         let width = term_width();
-        let diff = TextDiff::configure()
-            .algorithm(Algorithm::Patience)
-            .timeout(Duration::from_millis(500))
-            .diff_lines(old.as_str(), new.as_str());
         print_line(width);
 
         if self.show_info {
             self.print_info();
         }
 
-        if !old.is_empty() {
+        let old_contents = self.old_snapshot.as_ref().map(|o| o.contents());
+        let new_contents = self.new_snapshot.contents();
+
+        if let Some(SnapshotContents::Binary(contents)) = old_contents {
             println!(
                 "{}",
-                style(format_args!("-{}", self.old_snapshot_hint)).red()
+                style(format_args!(
+                    "-{}: {}",
+                    self.old_snapshot_hint,
+                    encode_file_link_escape(&contents.build_path(self.snapshot_file.unwrap())),
+                ))
+                .red()
             );
         }
-        println!(
-            "{}",
-            style(format_args!("+{}", self.new_snapshot_hint)).green()
-        );
 
-        println!("────────────┬{:─^1$}", "", width.saturating_sub(13));
-        let mut has_changes = false;
-        for (idx, group) in diff.grouped_ops(4).iter().enumerate() {
-            if idx > 0 {
-                println!("┈┈┈┈┈┈┈┈┈┈┈┈┼{:┈^1$}", "", width.saturating_sub(13));
+        if let SnapshotContents::Binary(contents) = new_contents {
+            println!(
+                "{}",
+                style(format_args!(
+                    "+{}: {}",
+                    self.new_snapshot_hint,
+                    encode_file_link_escape(
+                        &contents
+                            .build_path(self.snapshot_file.unwrap().with_extension("snap.new"))
+                    ),
+                ))
+                .green()
+            );
+        }
+
+        if let Some((old, new)) = match (
+            self.old_snapshot.as_ref().map(|o| o.contents()),
+            self.new_snapshot.contents(),
+        ) {
+            (Some(SnapshotContents::Binary(_)) | None, SnapshotContents::Text(new)) => {
+                Some((None, Some(new.to_string())))
+            }
+            (Some(SnapshotContents::Text(old)), SnapshotContents::Binary { .. }) => {
```

Comment:
Presumably we should never we comparing these?

(No great problem, though it also makes me think whether we can overall simplify this sort of code...)

Reply:
Well, you can change a text snapshot to a binary one and vice versa since they essentially live in the same namespace. Also I did find that feature somewhat helpful when switching some things to binary snapshots where previously I had a snapshoted a struct with measurements.

I suppose the alternatives are to either just show something like `- old text snapshot` in red or show nothing at all (like would be the case with inline snapshots, where in order to replace it you have to delete the old snapshot in the code), but I would prefer to show the old contents somehow and I think the way this displays currently is quite reasonable:

![image](https://github.com/user-attachments/assets/ece82450-34c7-4189-afdc-d98447bd2634)

But I'm definitely open to suggestions for a nicer UI for this.

Reply:
Ah yes, good point, I hadn't thought of that. 

(this would be the sort of thing that would be great to add an integration test — i.e. what happens if we switch between text and binary...)